test/cql-pytest/test_compaction_tombstone_gc.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+from util import new_materialized_view, new_test_table
+import nodetool
+import time
+
+def test_tombstone_gc_with_conflict_in_memtable(scylla_only, cql, test_keyspace):
+    """
+    Regression test for fixed https://github.com/scylladb/scylladb/issues/20423
+    """
+    schema = "k int, v int, primary key (k, v)"
+    with new_test_table(cql, test_keyspace, schema, extra="with gc_grace_seconds = 0") as table:
+        with nodetool.no_autocompaction_context(cql, table):
+            nodetool.setlogginglevel(cql, "compaction", "trace")
+            # Insert initial data into the base table:
+            # Row 1 is expected to be garbage collected after expiration
+            cql.execute(f"insert into {table} (k, v) values (1, 1) using timestamp 1 and ttl 1")
+            cql.execute(f"insert into {table} (k, v) values (1, 2) using timestamp 2")
+            nodetool.flush(cql, table)
+            # Delete row 2, this deletion is expected to be kept
+            # when we insert backdated live data into the memtable
+            cql.execute(f"delete from {table} using timestamp 3 where k=1 and v=2")
```

Comment:
Since gc_grace_seconds=0, in theory this deletion can be gc'ed immediately. What prevents this from happening, the "reordering grace period" variable (I forgot its name)? Maybe you should mention that.

Reply:
gc_grace_seconds=0 still requires a single second to pass.
maybe it's some `>` somewhere vs. `>=`.

Reply:
I'm very surprised now. I was always assuming that gc_grace_seconds=0 meant an immediate gc, not one second. I agree that for dtests and sct and the likes, nobody cares about one second, but why did this happen? Why does setting the grace period to 0 still means waiting one second? Was this intentional? Is it documented anywhere?

Reply:
We now also have tombstone_gc_mode::immediate by the way.

@raphaelsc?